### PR TITLE
Warn validators with slow hardware

### PR DESCRIPTION
**Pull request by Szegoo**
Companion for: [#12620](https://github.com/paritytech/substrate/pull/12620)

cumulus companion: https://github.com/paritytech/cumulus/pull/1863

### CODE REVIEW

**Comment by ordian** [comment]
I think it's fine to enable hw bench if `--validator` is passed, but it should be possible to disable it even with `--validator` with explicit flag.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2289437

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2292416

**Review by ggwpez — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]

**Comment by ggwpez** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2294127

**Comment by ggwpez** [comment]
bot merge

**Comment by Szegoo** [comment]
It is weird to see that Cumulus is still failing the CI check.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2295083

File: cli/src/command.rs [hunk 1]
```diff
@@ -328,14 +328,12 @@ where
 	};
 
 	runner.run_node_until_exit(move |config| async move {
-		let hwbench = if !cli.run.no_hardware_benchmarks {
-			config.database.path().map(|database_path| {
+		let hwbench = (!cli.run.no_hardware_benchmarks)
+			.then_some(config.database.path().map(|database_path| {
 				let _ = std::fs::create_dir_all(&database_path);
 				sc_sysinfo::gather_hwbench(Some(database_path))
```
Comment (mordamax) [inline cli/src/command.rs:334]: shouldn't be there 3 args?
https://github.com/paritytech/substrate/pull/12620/files#diff-933dca0c14add2a08eaca93fe746ada8e72f547d2861ca2079f191908984f06fR594

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2295538

File: cli/src/command.rs (hunk 1 above)
Comment (ggwpez) [reply cli/src/command.rs:334]: Yea but i am un-doing that here now https://github.com/paritytech/substrate/pull/13198  
I will ping again when the comp check fails

**Comment by ggwpez** [comment]
bot merge

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**